1845 - IdsDatagrid Part 1 of converted tests

andrivarhanov opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**
Datagrid has a lot of tests so decided to split them into several PRs

**Related github/jira issue (required)**:
#1845 

**Steps necessary to review your pull request (required)**:
- tests should pass
- coverage increased

**Some failed tests/functionality to consider**

~~1. `addRows` doesn't work on editable datagrid~~
~~go to http://localhost:4300/ids-data-grid/editable.html~~
~~test `can add multiple rows at given index` should pass after the fix, skipped at the moment~~

---

~~2. `selectRow` and `deSelectRow` produces console error on invalid index~~
~~- test `has no errors on invalid selectRow / deSelectRow calls` should pass after the fix, skipped at the moment~~

---
3. it doesn't trigger cell hyperlinks on Enter anymore
- navigate to a cell with link with keyboard and press Enter
---
4. it doesn't update `dataGrid.activeCell` when navigating with keyboard and doesn't trigger `activecellchanged` event
- run it in the console `$('ids-data-grid').addEventListener('activecellchanged', (e) => console.log(e.detail.activeCell))` and navigate cell using arrow up/right/bottom/left

**Included in this Pull Request**:
- [ ] Some documentation for the feature.
- [x] A test for the bug or feature.
- [ ] A note to the change log.

<!-- After submitting your PR, please check back to make sure checks on the PR -->

jdilag90 commented:
Hey @andrivarhanov 

`addRows()` will work if it's passed a dataset with `book` properties, `book` is configured as the primary key which datasource uses to find/update row data.
Might help to get those tests to pass.

```
$('ids-data-grid').addRows([
  { description: 'test1', book: 'test1' },
  { description: 'test2', book: 'test2' },
  { description: 'test3', book: 'test3' }
], 2)
```

tmcconechy commented:
@andrivarhanov 

1. Answered by @jdilag90 
2. Can you fix that one as well? If its quick?
3. Might need to look into this later. Fine to skip for now
4. Might need to look into this later. Fine to skip for now

Can you log a bug for the ones you dont get to? Could all be on one bug and i can look later.

andrivarhanov commented:
@jdilag90 thanks for the help, it works now

@tmcconechy 
1. fixed
2. fixed

sure, I will create separated issue for all failed tests/functionality
